src/js/actions/tools.js

```diff
@@ -327,13 +332,19 @@ define(function (require, exports) {
                 return currentTool.deselectHandler.call(this, currentTool);
             })
             .then(function () {
-                var psToolName = nextTool.nativeToolName,
+                var psToolName = nextTool.nativeToolName.call(this),
                     setToolPlayObject = toolLib.setTool(psToolName);
 
                 // Set the new native tool
                 return descriptor.playObject(setToolPlayObject);
             })
             .then(function () {
+                if (!nextTool.handleVectorMaskMode) {
```

Comment:
I disagree, because we now have 2 distinct subgroups of tools that either can handle vector masks or not. the behavior that we want is to drop out of vector mask editing mode when we swtich to a tool ( type or sampler currently ) that does not support editing vector modes, I do not think that modifying each new tool to drop out of vector mask mode on select is as maintainable a solution as dropping out of vector mask mode here.